app/models/manageiq/providers/autosde/storage_manager/storage_service.rb

```diff
@@ -3,10 +3,15 @@ class ManageIQ::Providers::Autosde::StorageManager::StorageService < ::StorageSe
   supports :delete
 
   def self.raw_create_storage_service(ext_management_system, options = {})
+    capability_value_list = []
+    ext_management_system.capabilities.each_key do |capability|
+      capability_value_list.push(options[capability]) unless options[capability] == '-1'
+    end
+
```

Comment:
You can replace most of this loop with a simple `slice().values`:  `options.slice(ext_management_system.capabilities.keys).values` then just `.delete("-1")`

```suggestion
    capability_value_list = options.slice(ext_management_system.capabilities.keys).values
    capability_value_list.delete("-1")

```

Reply:
hi @agrare, 
I changed as suggested
thanks

Reply:
thanks, they're co-dependent, as the ui generates fields from the provider's params for create
added to both descriptions